deployment/src/main/java/io/quarkiverse/jackson/jq/deployment/JacksonJqProcessor.java

```diff
@@ -1,59 +1,146 @@
 import java.util.List;
-import java.util.Map;
 
 import javax.inject.Singleton;
 
+import org.apache.commons.lang3.StringUtils;
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+
 import io.quarkiverse.jackson.jq.JacksonJqScopeRecorder;
+import io.quarkiverse.jackson.jq.JqFunction;
 import io.quarkus.arc.deployment.SyntheticBeanBuildItem;
+import io.quarkus.arc.deployment.UnremovableBeanBuildItem;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.ExecutionTime;
 import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.AdditionalApplicationArchiveMarkerBuildItem;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
-import io.quarkus.deployment.builditem.nativeimage.NativeImageResourcePatternsBuildItem;
-import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 import io.quarkus.deployment.recording.RecorderContext;
+import io.quarkus.runtime.RuntimeValue;
+import net.thisptr.jackson.jq.BuiltinFunction;
 import net.thisptr.jackson.jq.BuiltinFunctionLoader;
 import net.thisptr.jackson.jq.Function;
 import net.thisptr.jackson.jq.Scope;
-import net.thisptr.jackson.jq.Versions;
+import net.thisptr.jackson.jq.VersionRange;
 
 class JacksonJqProcessor {
-
     private static final String FEATURE = "jackson-jq";
-    private static final String REFLECTION_VERSION_RANGE_DESERIALIZER = "net.thisptr.jackson.jq.internal.misc.VersionRangeDeserializer";
-    private static final String JSON_CONFIG_GLOB = "net/thisptr/jackson/**/*.json";
 
     @BuildStep
     FeatureBuildItem feature() {
         return new FeatureBuildItem(FEATURE);
     }
 
     @BuildStep
-    List<ReflectiveClassBuildItem> registerForReflection() {
-        final List<ReflectiveClassBuildItem> reflectionConfig = new ArrayList<>();
-        reflectionConfig.add(ReflectiveClassBuildItem.weakClass(REFLECTION_VERSION_RANGE_DESERIALIZER));
-        return reflectionConfig;
-    }
+    @Record(ExecutionTime.STATIC_INIT)
+    SyntheticBeanBuildItem quarkusScopeBean(
+            CombinedIndexBuildItem combinedIndex,
+            JacksonJqConfig config,
+            JacksonJqScopeRecorder recorder,
+            List<JacksonJqFunctionBuildItem> functions,
+            RecorderContext context) throws Exception {
 
-    @BuildStep
-    NativeImageResourcePatternsBuildItem includeJsonConfigFile() {
-        return NativeImageResourcePatternsBuildItem.builder().includeGlob(JSON_CONFIG_GLOB).build();
-    }
+        IndexView indexView = combinedIndex.getIndex();
+        List<String> excludes = config.functions.excludes.orElseGet(Collections::emptyList);
+
+        RuntimeValue<Scope> root = recorder.createScope();
+        RuntimeValue<Scope> local = recorder.createScope(root);
```

Comment:
Two scope will be created here:
1. root -> contains all the build in functions
2. local -> contains all the local functions and has `root` as parent. This is done to keep a separation between builtin and local functions os you may also want to override some built in functions if needed.